smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -139,7 +144,26 @@ void execute() {
         // Generate models that are connected to the service being generated.
         LOGGER.fine("Walking shapes from " + service.getId() + " to find shapes to generate");
         Set<Shape> serviceShapes = new TreeSet<>(new Walker(nonTraits).walkShapes(service));
-        serviceShapes.forEach(shape -> shape.accept(this));
+        Map<String, Shape> shapeMap = new TreeMap<>();
```

Comment:
I think moving this into its own method would make it more readable. You could then just return the created shape map.

Reply:
Fixed in cf9d401